packages/components/src/index.tsx

```diff
@@ -35,29 +44,25 @@ const CourierSdk: React.FunctionComponent<{
 }> = ({ activeComponents, children }) => {
   const courier = useCourier();
 
-  useEffect(() => {
-    const courierSdk: any = {};
+  if (!window.courierSdk.transport) {
+    window.courierSdk.transport = courier.transport;
+  }
 
+  useEffect(() => {
     for (const component of Object.keys(activeComponents)) {
-      if (!courier[component]) {
+      const typedComponent = component as "inbox" | "toast";
+
+      if (!courier[typedComponent] || window.courierSdk[typedComponent]) {
         return;
       }
 
-      courierSdk[component] = courier[component];
-    }
-
-    courierSdk.transport = courier.transport;
-    window.courierSdk = courierSdk;
-    if (!window.courierAsyncInit) {
-      return;
-    }
-
-    for (const init of window.courierAsyncInit) {
-      init();
+      window.courierSdk[typedComponent] = courier[typedComponent];
+      const initActions = actions[`${typedComponent}/init`] ?? [];
+      for (const initAction of initActions) {
+        initAction();
+      }
     }
-
-    window.courierAsyncInit = [];
-  }, [courier]);
+  }, [courier, activeComponents]);
```

Comment:
I wonder the side effects of this. since active components is an object if you only add the toast, and then add the inbox later seems like it wouldn't cause a re-render

Reply:
yeah, thats where we need to mutation observer